pkg/util/mount/mount_linux.go

```diff
@@ -430,3 +430,40 @@ func (mounter *SafeFormatAndMount) getDiskFormat(disk string) (string, error) {
 	// and MD RAID are reported as FSTYPE and caught above).
 	return "unknown data, probably partitions", nil
 }
+
+// IsSamefsGroup is called only for requests to mount an already mounted
+// volume. It checks if fsGroup of new mount request is the same or not.
+// It return false if it not the same.
+func IsSamefsGroup(dir string, fsGroup *int64) (bool, error) {
+	s := &syscall.Stat_t{}
+	if err := syscall.Stat(dir, s); err != nil {
+		return false, err
+	}
+	gidNew := 0
+	if fsGroup != nil {
+		gidNew = int(*fsGroup)
+	}
+	glog.V(4).Infof("######### Old Gid %d New Gid %d  ###########", int(s.Gid), gidNew)
+	return int(s.Gid) == gidNew, nil
+}
+
+// IsAlreadyMounted is called to check if a requested volume has already been
+// mounted. It returns nil if no previous mount was found, otherwise it returns
+// the path to existing mount.
+func IsAlreadyMounted(dir string) (string, error) {
```

Comment:
@msau42 Since it is the end of day here on east coast and my head is not working well ;)
we can check fsgroup against already existing mount point, right?  local.go we have only where kubelet wants to bind mount a volume. So unless we find where the volume has already been mounted I do not see how we can compare fsGroup user wants with possibly already existing fsGroup. Please clarify how you suggest to check fsGroup without checking for existing mount points..